contrib/babelfishpg_tsql/sql/upgrades/babelfishpg_tsql--3.0.0--3.1.0.sql

```diff
@@ -1100,6 +1100,109 @@ AND sys.babelfish_get_pltsql_function_signature(ao.object_id) = f.funcsignature
 WHERE ao.type in ('P', 'RF', 'V', 'TR', 'FN', 'IF', 'TF', 'R');
 GRANT SELECT ON sys.all_sql_modules_internal TO PUBLIC;
 
+-- deprecate old FOR XML/JSON functions if they exist - if this is a fresh install of 3.1 they won't exist, so check for that
+DO $$
+BEGIN IF (SELECT count(*) FROM pg_proc as p where p.proname = 'tsql_query_to_xml') = 1 THEN
+    ALTER FUNCTION sys.tsql_query_to_xml(text, int, text, boolean, text) RENAME TO tsql_query_to_xml_deprecated_in_2_4_0;
+    CALL sys.babelfish_drop_deprecated_object('function', 'sys', 'tsql_query_to_xml_deprecated_in_2_4_0');
+END IF;
+END $$;
+
+DO $$
+BEGIN IF (SELECT count(*) FROM pg_proc as p where p.proname = 'tsql_query_to_xml_text') = 1 THEN
+    ALTER FUNCTION sys.tsql_query_to_xml_text(text, int, text, boolean, text) RENAME TO tsql_query_to_xml_text_deprecated_in_2_4_0;
```

Comment:
proper name would be `tsql_query_to_xml_text_deprecated_in_3_1_0`

Reply:
renamed to 3_1_0